mlx/backend/metal/kernels/quantized.h

```diff
@@ -0,0 +1,1461 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include <metal_simdgroup>
+#include <metal_stdlib>
+
+#include "mlx/backend/metal/kernels/bf16.h"
+#include "mlx/backend/metal/kernels/defines.h"
+#include "mlx/backend/metal/kernels/utils.h"
+
+#include "mlx/backend/metal/kernels/steel/gemm/gemm.h"
+
```

Comment:
So the way I've been trying to do these includes is to put them in the `.metal` file instead. For the no JIT case it should work as long as they are above the `quantized.h` include. Then for the JIT case what you do is use the JIT include strings when building the source string to compile. These should all be available already. That way we can avoid duplicating all that source code for the JIT case.

Reply:
Makes total sense -- I've updated it